File: AkitaModel/PersonBase.cs
namespace Skyware.Lis.AkitaModel;

/// <summary>
/// Abstract class for people.
/// </summary>
public abstract class PersonBase
{

    /// <summary>
    /// PK.
    /// </summary>
    public int? Id { get; set; }

    /// <summary>
    /// Given (first) name.
    /// </summary>
    public string GivenName { get; set; } = string.Empty;

    /// <summary>
    /// Middle (second) name.
    /// </summary>
    public string MiddleName { get; set; }

    /// <summary>
    /// Family name (surname, last name).
    /// </summary>
    public string FamilyName { get; set; } = string.Empty;

}

File: AkitaModelJava/akitamodel/src/main/java/skyware/lis/akitamodel/PersonBase.java
package skyware.lis.akitamodel;

import lombok.Data;

/**
 * Abstract class for people.
 */
@Data
public abstract class PersonBase {

    /**
     * Unique identifier (PK).
     */
    private Integer id;

    /**
     * Given (first) name.
     */
    private String givenName = "";

    /**
     * Middle (second) name.
     */
    private String middleName;

    /**
     * Family (last) name (Surname).
     */
    private String familyName = "";

}
